### ISSUE TITLE
usage of gulp-tslint breaks gulp-typescript with ts 1.5

### ISSUE THREAD

**Issue by maku** [panuhorsmalahti/gulp-tslint#21]
I use gulp-typescript with the latest 1.5 beta (https://github.com/mhegazy/TypeScript) . When I install gulp-tslint it does not work anymore. 

when require('gulp-tslint') before require('gulp-typescript') than I get then error:

```
var scanner = ts.createScanner(1, false, node.text);
                         ^
TypeError: undefined is not a function
```

the other way (gulp-typescript before gulp-tslint) I get an error on typescript transpiling : 

```
TypeError: object is not a function
```

Is there a workaround?

**Comment by panuhorsmalahti** [comment]
Are you using typescript@1.50-beta from https://www.npmjs.com/package/typescript or that fork?

It seems that the problem is that gulp-typescript uses typescript 1.4.1, whereas tslint/gulp-tslint uses 1.5.0-beta.

What's the version of typescript in node_modules/typescript? And is the typescript under gulp-typescript (e.g. node_modules/gulp-typescript/node_modules/typescript), and what's that version?

**Comment by maku** [comment]
As I mentioned I use latest typescript from https://github.com/mhegazy/TypeScript which is v1.5-beta2 (the fork)
gulp-typescript is configured so that it uses this version...

**Comment by panuhorsmalahti** [comment]
I don't think gulp-tslint should support unofficial TypeScript forks (assuming that is the cause of the error here).

**Comment by maku** [comment]
maybe, but I would nice when it would be configurable (like gulp-typescript). So it would be possible to work with a more recent version. I think especially in the future this would be comfortable.

**Comment by panuhorsmalahti** [comment]
I agree. The thing is, I don't know the cause of this issue, and I don't know the structure of your node_modules. Can you list every "typescript" folder in node_modules (recursively), and each version so that I can understand what typescript is used and where? In addition, you could try installing typescript@1.5.0-beta manually to e.g. node_modules/gulp-tslint/node_modules.

**Comment by panuhorsmalahti** [comment]
This should probably be closed now that TypeScript 1.5 is released, tslint is released for it and so is gulp-tslint.
